pkg/cockroach/cockroach_test.go

```diff
@@ -1 +1,98 @@
 package cockroach
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestBuildDSN(t *testing.T) {
+	cases := []struct {
+		name   string
+		params map[string]string
+		want   string
+	}{
+		{
+			name: "valid URI",
+			params: map[string]string{
+				"host":     "localhost",
+				"port":     "26257",
+				"user":     "root",
+				"ssl_mode": "enable",
+				"ssl_dir":  "/tmp",
+			},
+			want: "application_name=dss host=localhost pool_max_conns=4 port=26257 sslcert=/tmp/client.root.crt sslkey=/tmp/client.root.key sslmode=enable sslrootcert=/tmp/ca.crt user=root",
+		},
+		{
+			name: "missing host",
+			params: map[string]string{
+				"port":     "26257",
+				"user":     "root",
+				"ssl_mode": "enable",
+				"ssl_dir":  "/tmp",
+			},
+			want: "",
```

Comment:
Doesn't test errors. A pattern that works with go style table based tests is to have one test to test success cases and on e test to test error cases.